1_fetch/src/maps_to_sf.R

```diff
@@ -0,0 +1,121 @@
+# Utils for using spatial data from `maps`
+# to create state and county sf objects
+
+generate_conus_sf <- function(proj_str) {
+  
+  usa_sf <- maps::map("usa", fill = TRUE, plot=FALSE) %>%
+    st_as_sf() %>% 
+    st_transform(proj_str) %>% 
+    st_buffer(0) 
+  
+  return(usa_sf)
+}
+
+generate_conus_states_sf <- function(proj_str) {
+  
+  usa_sf <- maps::map("usa", fill = TRUE, plot=FALSE) %>%
+    st_as_sf() %>% 
+    st_transform(proj_str) %>% 
+    st_buffer(0) 
+  
+  # Need to remove islands from state outlines and then add back in 
+  # later so that they can be drawn as individual polygons. Otherwise,
+  # drawn with the state since the original state maps data only has 1
+  # ID per state.
+  
+  usa_islands_sf <- usa_sf %>% filter(ID != "main")
+  usa_addl_islands_sf <- generate_addl_islands(proj_str)
+  usa_mainland_sf <- usa_sf %>% 
+    filter(ID == "main") %>% 
+    st_erase(usa_addl_islands_sf) 
+  
+  # Have to manually add in CO because in `maps`, it is an incomplete
+  # polygon and gets dropped somewhere along the way.
+  co_sf <- maps::map("state", "colorado", fill = TRUE, plot=FALSE) %>%
+    st_as_sf() %>%
+    # st_buffer(0) %>% # Hmm thought it would fix the weird line but doesn't 
+    st_transform(proj_str)
+  
+  maps::map("state", fill = TRUE, plot=FALSE) %>%
+    st_as_sf() %>%
+    st_transform(proj_str) %>%
+    st_buffer(0) %>% 
+    # Get rid of islands from state outline data
+    st_intersection(usa_mainland_sf) %>%
+    select(-ID.1) %>% # st_intersection artifact that is unneeded
+    # Add islands back in as separate polygons from states
+    bind_rows(usa_islands_sf) %>%
+    bind_rows(usa_addl_islands_sf) %>% 
+    st_buffer(0) %>%
+    st_cast("MULTIPOLYGON") %>% # Needed to bring back to correct type to use st_coordinates
+    rmapshaper::ms_simplify(0.5) %>%
+    bind_rows(co_sf) # bind CO after bc otherwise it gets dropped in st_buffer(0)
+  
+}
+
+generate_addl_islands <- function(proj_str) {
```

Comment:
It was more about how they are drawn once you are creating the SVG. I had to split up like this to make sure the islands are not connected together when drawing. There is a ton of weirdness. It would be fun to explore a different package, but this `maps` package is what we've always used, so I had the code to get around the funkiness already :) 

I know that there is also [`usmap`](https://usmap.dev/) (no territories, so would still need to add those) which I've wanted to explore. Creating an issue to do this, see #6

Reply:
Ah, I see. That makes sense. So much to think through! Totally makes sense to use existing code that has all the workarounds.